### ISSUE TITLE
 Can Orion handle training a 2TB dataset?

### ISSUE THREAD

**Issue by bigmisspanda** [sintel-dev/Orion#567]
* Orion version:latest
* Python version:3.11
* Operating System:Linux

### Description

In my case, the training data is very large and cannot be loaded into memory all at once. It seems that time_segments_aggregate, SimpleImputer, MinMaxScaler, and rolling_window_sequences in the pipeline all require the data to be stored in memory. Can Orion handle training a 2TB dataset?"

**Comment by sarahmish** [comment]
Hi @bigmisspanda – thank you for your question!

You are right, all the preprocessing primitives require to be in memory. 

One work around can be to replace these primitives with your own scalable functions and then start the Orion pipeline from the modeling primitive directly. Another can be to chunk up your training data and training the pipeline on each chunk.

**Comment by bigmisspanda** [comment]
> Hi @bigmisspanda – thank you for your question!
> 
> You are right, all the preprocessing primitives require to be in memory.
> 
> One work around can be to replace these primitives with your own scalable functions and then start the Orion pipeline from the modeling primitive directly. Another can be to chunk up your training data and training the pipeline on each chunk.

Yes, I understand what you mean. My plan is to use `TadGAN `to train an anomaly detection model. My data comes from power equipment sensors and has over 20 features. If I train in chunks, the `MinMaxScaler `results will not be globally distributed. I referred to the information in this [document,](https://medium.com/mit-data-to-ai-lab/time-series-anomaly-detection-in-the-era-of-deep-learning-f0237902224a) 
![image](https://github.com/user-attachments/assets/ba2b29ce-a183-4092-bd91-6489615a079f)
and my plan is: 1. To use MinMaxScaler's `partial_fit `for global calculations on the dataset in advance. 2. To split the data into chunks. 3. To remove the `MinMaxScaler `from the third step in the primitives. 4. To train the model. Is my approach feasible? Can `TadGAN `perform similar partial_fit training?

**Comment by sarahmish** [comment]
Your plan looks logical to me!

I'm not too familiar with what `partial_fit` does under the hood, however, calling `fit` on multiple times on different data chunks seems analogous to their concept of "incremental learning".
